logger/main.go

```diff
@@ -76,12 +80,6 @@ func main() {
 			if resp != nil && resp.Node != nil {
 				drainChan <- resp.Node.Value
 			}
-			time.Sleep(time.Duration(publishInterval) * time.Second)
-		}
-	}()
-
-	for {
-		select {
 		case <-signalChan:
 			close(exitChan)
 			stopChan <- true
```

Comment:
You should also stop the ticker. Github doesn't allow me to add a comment on a line that is hidden.

```
 		case <-cleanupChan:
 			return
```

To

```
 		case <-cleanupChan:
 			t.Stop()
 			return
```

to stop the ticket (avoid leaking goroutines)